src/maven/background_peaks_update.cpp

```diff
@@ -214,10 +214,15 @@ void BackgroundPeakUpdate::processSlices(vector<mzSlice*>&slices, string setName
 
             matchFragmentation(&group);
 
+            //Issue 21: if a non-null compound could not be associated with a group during matchFragmentation(),
+            //skip over the peak group.
+            if (!group.compound && compound){
+                continue;
+            }
+
```

Comment:
Previously, if no compound could be found for a peakgroup given thresholds, the compound would be associated anyway, but with a score of `0` instead of the fragmentation score.
Now, the matches are just skipped.